pkg/monitoring/metrics/forklift-controller/migration_metrics.go

```diff
@@ -68,51 +66,51 @@ func RecordMigrationMetrics(c client.Client) {
 
 				provider := sourceProvider.Type().String()
 
-				if m.Status.HasCondition(Succeeded) {
-					key = fmt.Sprintf("%s|%s|%s|%s|%s", Succeeded, provider, mode, target, string(plan.UID))
-					counterMap[key]++
+				statusList := []string{Succeeded, Failed, Executing, Canceled}
+				for _, status := range statusList {
+					if m.Status.HasCondition(status) {
+						switch status {
+						case Succeeded:
+							if _, exists := processedSucceededMigrations[string(m.UID)]; !exists {
+								migrationStatusCounter.With(prometheus.Labels{"status": Succeeded, "provider": provider, "mode": mode, "target": target}).Inc()
+								migrationPlanCorrelationStatuCounter.With(prometheus.Labels{"status": Succeeded, "provider": provider, "mode": mode, "target": target, "plan": string(plan.UID)}).Inc()
```

Comment:
this part exists in all cases, how about doing it once? if that's not possible, how about pulling it out to a function?

Reply:
It's not possible to use one since we are interested in specific statuses, but it has been extracted to a function as you suggested.